include/veriblock/blockchain/alt_block_addon.hpp

```diff
@@ -25,94 +25,110 @@ struct AltBlockAddon : public PopState<AltEndorsement> {
   // remove this
   uint8_t chainWork;
 
-  //! list of changes introduced in this block
-  // ATV::id_t
-  std::vector<uint256> atvids;
-  // VTB::id_t
-  std::vector<uint256> vtbids;
-  // VbkBlock::id_t
-  std::vector<uint96> vbkblockids;
-
   bool payloadsIdsEmpty() const {
-    return atvids.empty() && vtbids.empty() && vbkblockids.empty();
+    return _atvids.empty() && _vtbids.empty() && _vbkblockids.empty();
   }
 
   bool operator==(const AltBlockAddon& o) const {
-    return atvids == o.atvids && vtbids == o.vtbids &&
-           vbkblockids == o.vbkblockids &&
+    return _atvids == o._atvids && _vtbids == o._vtbids &&
+           _vbkblockids == o._vbkblockids &&
            PopState<AltEndorsement>::operator==(o);
   }
 
   template <typename pop_t, typename pop_id_t>
-  std::vector<pop_id_t>& getPayloadIds();
+  const std::vector<pop_id_t>& getPayloadIds() const;
 
   template <typename pop_t, typename pop_id_t>
-  const std::vector<pop_id_t>& getPayloadIds() const;
+  void removePayloadIds(const pop_id_t& pid) {
+    auto& payloads = getPayloadIdsInner<pop_t, pop_id_t>();
+    auto it = std::find(payloads.begin(), payloads.end(), pid);
+    VBK_ASSERT(it != payloads.end());
+    payloads.erase(it);
+  }
 
-  void setNull() {
-    PopState<AltEndorsement>::setNull();
-    chainWork = 0;
-    atvids.clear();
-    vtbids.clear();
-    vbkblockids.clear();
+  template <typename pop_t, typename pop_id_t>
+  void insertPayloadIds(const pop_id_t& pid) {
```

Comment:
Either make this method singular, e.g. `insertPayloadId` or accept vector of ids (or both, as 2 separate methods)